scripts/addheaders.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Copyright 2023 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if ! hash addlicense 2>/dev/null; then
+  go install github.com/google/addlicense@latest
+fi
+
+# adds a new line between the header and the package keyword when it does not exists.
+find . -type f -name "*.go" -exec sed -i '/\/\/ limitations under the License\.$/{N; s/\(\/\/ limitations under the License.\)\n\(package\)/\1\n\n\2/}' {} \;
```

Comment:
`sed -i` behaves differently on MacOS we may have to revise this, but I have not tested it yet

Reply:
oh, I missed it.

Reply:
yeah, that's right, I didn't spot it because I've replaced the built-in sed with GNU sed to avoid problem with other projects that uses `sed`.

I do remember that we need to add .bak or something similar to get it working with the built-in one.

Will take a look and update it.

Reply:
okay, did a small update on the command, tested on macos and fedora, both working.
@ckadner if you have time, please test it on your machine as well.

1- remove the space between the header and the package keyword for any .go file
2- run `make fmt`
3- the changed file should have the space added again.

PS: the addlicense will add the header only on new files.